UML-Diagram/ReportSystem.puml

```diff
@@ -1,32 +1,28 @@
 @startuml ReportSystem
-package "Report System" {
-    
-    ' Define main abstractc
-    abstract class ABorrowReport implements IReporting{
-        - BorrowManager : BorrowList
-    }
 
-    abstract class APatronReport implements IReporting{
-        - PatronManager : PatronList
+package "Report System" {
+    ' Define main contract
+    interface IReport {
+        + GenerateReport() : void
     }
 
-    ' Define main interface
-    interface IReporting {
-        - LogService : ILogService
-        + generateReport()
+    ' Define main abstract base
+    abstract class BaseReport implements IReport {
+        - BorrowManager : _borrowManager
     }
 
     ' Define main classes
-    class BookCurrentBorrwedReport extends ABorrowReport {
-        + BookBorrowedReportFormat()
+    class BorrowPatronHistory extends BaseReport {
+        + PatronBorrowHistoryReport() : void
     }
 
-    class OverdueBooksReport extends ABorrowReport {
-        + OverdueBooksFormat()
+    class CurrentBorrowBooksReport extends BaseReport {
+        + BorrowedBooksReport() : void
     }
 
-    class BorrowingHistoryReport extends APatronReport {
-        + BorrowHistoryFormat()
+    class OverdueBooksReport extends BaseReport {
+        + OverdueBooksListReport() : void
```

Comment:
Name of methods its different from the interface

Reply:
Can you review the code we have on **develop** branch to verify please?

For example, `OverdueBooksReport` class has this code, which I represent on the UML and interface method it's inside of it due to we're using for printing a message in the console.
```
namespace ReportSystem.Concrete;
public class OverdueBooksReport : BookReport
{
    public OverdueBooksReport(BorrowManager BorrowList) : base(BorrowList){}

    public void OverdueBooksListReport()
    {
        generateReport();
        foreach(Borrow borrow in BorrowList.GetBorrows()){
            if(DateTime.Now > borrow.GetDueDate()){
                borrow.BorrowDetails();
            }
        }        
    }
}
```

Reply:
i reviewed and yeah can be managed that way